ESP_ATMod/ESP_ATMod.ino

```diff
@@ -76,57 +80,58 @@ const char APP_VERSION[] = "0.2.8";
+					   {nullptr, TYPE_NONE, 0, 0}};
 
 uint8_t sendBuffer[2048];
-uint16_t dataRead = 0;  // Number of bytes read from the input to a send buffer
+uint16_t dataRead = 0; // Number of bytes read from the input to a send buffer
 
 // TLS specific variables
 
-uint8_t fingerprint[20];  // SHA-1 certificate fingerprint for TLS connections
+uint8_t fingerprint[20]; // SHA-1 certificate fingerprint for TLS connections
 bool fingerprintValid;
-BearSSL::X509List *CAcert = nullptr;  // CA certificate for TLS validation
+BearSSL::X509List CAcert; // CA certificate for TLS validation
+size_t maximumCertificates;
 
-char *PemCertificate = nullptr;  // Buffer for loading a certificate
-uint16_t PemCertificatePos;  // Position in buffer while loading
-uint16_t PemCertificateCount;  // Number of chars read
+char *PemCertificate = nullptr; // Buffer for loading a certificate
+uint16_t PemCertificatePos;		// Position in buffer while loading
+uint16_t PemCertificateCount;	// Number of chars read
 
 /*
  *  Global settings
  */
-bool gsEchoEnabled = true;  // command ATE
-uint8_t gsCipMux = 0;  // command AT+CIPMUX
-uint8_t gsCipdInfo = 0;  // command AT+CIPDINFO
-uint8_t gsCwDhcp = 3;  // command AT+CWDHCP_CUR
-bool gsFlag_Connecting = false;  // Connecting in progress (CWJAP) - other commands ignored
-int8_t gsLinkIdReading = -1;  // Link id where the data is read
-bool gsCertLoading = false;  // AT+CIPSSLCERT in progress
-bool gsWasConnected = false;  // Connection flag for AT+CIPSTATUS
-uint8_t gsCipSslAuth = 0;  // command AT+CIPSSLAUTH: 0 = none, 1 = fingerprint, 2 = certificate chain
-uint8_t gsCipRecvMode = 0;  // command AT+CIPRECVMODE
-ipConfig_t gsCipStaCfg = { 0, 0, 0 };  // command AT+CIPSTA
-dnsConfig_t gsCipDnsCfg = { 0, 0 };  // command AT+CIPDNS
-uint16_t gsCipSslSize = 16384;  // command AT+CIPSSLSIZE
-bool gsSTNPEnabled = true;  // command AT+CIPSNTPCFG
-int8_t gsSTNPTimezone = 0;  // command AT+CIPSNTPCFG
-String gsSNTPServer[3];  // command AT+CIPSNTPCFG
+bool gsEchoEnabled = true;			// command ATE
+uint8_t gsCipMux = 0;				// command AT+CIPMUX
+uint8_t gsCipdInfo = 0;				// command AT+CIPDINFO
+uint8_t gsCwDhcp = 3;				// command AT+CWDHCP_CUR
+bool gsFlag_Connecting = false;		// Connecting in progress (CWJAP) - other commands ignored
+int8_t gsLinkIdReading = -1;		// Link id where the data is read
+bool gsCertLoading = false;			// AT+CIPSSLCERT in progress
+bool gsWasConnected = false;		// Connection flag for AT+CIPSTATUS
+uint8_t gsCipSslAuth = 0;			// command AT+CIPSSLAUTH: 0 = none, 1 = fingerprint, 2 = certificate chain
+uint8_t gsCipRecvMode = 0;			// command AT+CIPRECVMODE
+ipConfig_t gsCipStaCfg = {0, 0, 0}; // command AT+CIPSTA
+dnsConfig_t gsCipDnsCfg = {0, 0};	// command AT+CIPDNS
+uint16_t gsCipSslSize = 16384;		// command AT+CIPSSLSIZE
+bool gsSTNPEnabled = true;			// command AT+CIPSNTPCFG
+int8_t gsSTNPTimezone = 0;			// command AT+CIPSNTPCFG
+String gsSNTPServer[3];				// command AT+CIPSNTPCFG
 
 /*

```

Comment:
I had problems with compiling: `undeclared function checkCertificateDuplicates`. I know Arduino IDE gathers function prototypes from the .ino file (that I really hate) and maybe in the IDE it compiles fine. But my environment (Sloeber) doesn't compile. Please add the following prototype. It should not affect the Arduino IDE compilation.
```
/*
 * Local prototypes
 */
static bool checkCertificateDuplicates(BearSSL::X509List &importCertList);
```

Reply:
Ofcourse normally I add this, could not say why it isn’t there anymore because I certainly wrote this once :) Resolved aswell.

Reply:
I'm not seeing the change. Did you commit the changes?

Reply:
Sorry, not yet commited. Busy with something now, I’m commit this in a few minutes-hour

Reply:
no worries, I will continue tomorrow.